airbyte-lib/airbyte_lib/factories.py

```diff
@@ -48,20 +48,41 @@ def new_local_cache(
 
 def get_connector(
     name: str,
-    version: str = "latest",
-    config: Optional[Dict[str, Any]] = None,
+    version: str | None = None,
+    pip_url: str | None = None,
+    config: dict[str, Any] | None = None,
```

Comment:
No config validation happens in this function right? As a user I would expect an exception to be raised if my config was wrong. Could we call check if config is not none?

Reply:
the config can be set later on - it will be validated once it gets set.

There is no specific error for the case if the config is never set and a method that requires it (like check or read_all) is called - I will add that on a separate PR.